src/main/java/com/github/steveice10/mc/protocol/data/game/chunk/FlexibleStorage.java

```diff
@@ -69,14 +60,38 @@ public void set(int index, int value) {
             throw new IllegalArgumentException("Value cannot be outside of accepted range.");
         }
 
-        int bitIndex = index * this.bitsPerEntry;
-        int startIndex = bitIndex / 64;
-        int endIndex = ((index + 1) * this.bitsPerEntry - 1) / 64;
-        int startBitSubIndex = bitIndex % 64;
-        this.data[startIndex] = this.data[startIndex] & ~(this.maxEntryValue << startBitSubIndex) | ((long) value & this.maxEntryValue) << startBitSubIndex;
-        if(startIndex != endIndex) {
-            int endBitSubIndex = 64 - startBitSubIndex;
-            this.data[endIndex] = this.data[endIndex] >>> endBitSubIndex << endBitSubIndex | ((long) value & this.maxEntryValue) >> endBitSubIndex;
+        int cellIndex = (int) (index * divideMultiply + divideAdd >> 32L >> divideShift);
+        int bitIndex = (index - cellIndex * valuesPerLong) * bitsPerEntry;
+        data[cellIndex] &= ~(maxEntryValue << bitIndex) | (value & maxEntryValue) << bitIndex;
+    }
+
+    private static final int[] MAGIC_VALUES = {
```

Comment:
I generally put static constants at the top of the file before other fields, not the bottom.

Reply:
I generally also do that, but I have to agree with Redned on this one because it is actually quite long.